### PR TITLE
Separate the api into a (sub)module

### PR DESCRIPTION
**Pull request by zregvart**
This might be a solution not to trickle down transitive dependencies not used by ec-cli from the enterprise-contract-controller. This way we can distribute the api without any additional dependencies.

### CODE REVIEW

**Comment by zregvart** [comment]
There is a [branch](https://github.com/zregvart/ec-cli/tree/replace-ecc) on my fork of ec-cli that can be used to test this before it's merged.

**Comment by lcarva** [comment]
Any impact on infra-deployments? Probably not.

**Comment by lcarva** [comment]
CI failure when building a container image (wasn't aware there was one!):
```
go mod download: github.com/hacbs-contract/enterprise-contract-controller/api@v0.0.0-00010101000000-000000000000 (replaced by ./api): reading api/go.mod: open /workspace/api/go.mod: no such file or directory
subprocess exited with status 1
subprocess exited with status 1
error building at STEP "RUN go mod download": exit status 1
```

**Comment by zregvart** [comment]
> Any impact on infra-deployments? Probably not.

I don't think so, the outcome is a go.mod/go.sum files in a subdirectory, and infra is pointing to YAML files.

**Comment by zregvart** [comment]
> CI failure when building a container image (wasn't aware there was one!):

Yeah, it's trying to cache dependencies and rely only on having `go.mod`/`go.sum` in the root of the repository.

File: go.sum [hunk 1]
```diff
@@ -360,6 +372,9 @@ github.com/google/pprof v0.0.0-20201023163331-3e6fc7fc9c4c/go.mod h1:kpwsk12EmLe
 github.com/google/pprof v0.0.0-20201203190320-1bf35d6f28c2/go.mod h1:kpwsk12EmLew5upagYY7GY0pfYCcupk39gWOCRROcvE=
 github.com/google/pprof v0.0.0-20210122040257-d980be63207e/go.mod h1:kpwsk12EmLew5upagYY7GY0pfYCcupk39gWOCRROcvE=
 github.com/google/pprof v0.0.0-20210226084205-cbba55b83ad5/go.mod h1:kpwsk12EmLew5upagYY7GY0pfYCcupk39gWOCRROcvE=
+github.com/google/pprof v0.0.0-20210601050228-01bbb1931b22/go.mod h1:kpwsk12EmLew5upagYY7GY0pfYCcupk39gWOCRROcvE=
+github.com/google/pprof v0.0.0-20210609004039-a478d1d731e9/go.mod h1:kpwsk12EmLew5upagYY7GY0pfYCcupk39gWOCRROcvE=
+github.com/google/pprof v0.0.0-20210720184732-4bb14d4b1be1/go.mod h1:kpwsk12EmLew5upagYY7GY0pfYCcupk39gWOCRROcvE=
```
Comment (simonbaird) [inline go.sum:377]: I don't understand go deps well, so maybe a dumb question, but can we do a mod tidy and remove the duplicates?

**Review by simonbaird — APPROVED** [review]

File: go.sum (hunk 1 above)
Comment (zregvart) [reply go.sum:377]: They're not duplicate because of the difference in version. I can't tell why these entries have the same hash for different versions. If you run:

```
$ go mod graph  |grep ' github.com/google/pprof'
```

you'll see some of these, and then even more of these. The [`go.sum`](https://go.dev/ref/mod#go-sum-files) file has every dependency, being direct or indirect computed by finding the [minimal version selection](https://go.dev/ref/mod#minimal-version-selection), and this is that output.

In other words, this has already been `go mod tidy`-ed.
